### PR TITLE
✨Add COM Energieherkunft

### PR DESCRIPTION
**Pull request by hf-aschloegl**
Fixes https://github.com/Hochfrequenz/BO4E-python/issues/157

### CODE REVIEW

File: src/bo4e/com/energieherkunft.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# pylint: disable=too-few-public-methods
+@attr.s(auto_attribs=True, kw_only=True)
+class Energieherkunft(COM):
+    """
+    Abbildung einer Energieherkunft.
+    """
+
+    # required attributes
+    erzeugungsart: Erzeugungsart = attr.ib(validator=attr.validators.in_(Erzeugungsart))
+    anteil_prozent: Decimal = attr.ib(validator=attr.validators.instance_of(Decimal))
```
Comment (hf-kklein) [inline src/bo4e/com/energieherkunft.py:25]: docstrings

File: src/bo4e/com/energieherkunft.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import attr
+from marshmallow import fields, post_load
+from marshmallow_enum import EnumField  # type:ignore[import]
+
+from bo4e.com.com import COM, COMSchema
+from bo4e.enum.erzeugungsart import Erzeugungsart
+
+
+# pylint: disable=too-few-public-methods
```
Comment (hf-kklein) [inline src/bo4e/com/energieherkunft.py:16]: foo is a foo ;)

File: src/bo4e/com/energieherkunft.py (hunk 1 above)
Comment (hf-aschloegl) [reply src/bo4e/com/energieherkunft.py]: 733a531a5f7334b4906f87e298a09d1611b045f9

File: src/bo4e/com/energieherkunft.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+class Energieherkunft(COM):
+    """
+    Abbildung einer Energieherkunft.
+    """
+
+    # required attributes
+    #: Art der Erzeugung der Energie.
+    erzeugungsart: Erzeugungsart = attr.ib(validator=attr.validators.in_(Erzeugungsart))
+    #: Prozentualer Anteil der jeweiligen Erzeugungsart.
+    anteil_prozent: Decimal = attr.ib(validator=attr.validators.instance_of(Decimal))
```
[the remark this answers is not in the document]
Comment (hf-aschloegl) [reply src/bo4e/com/energieherkunft.py:27]: 1dcbc9f95ebeb3b3c37917e084f95e7343bf57e7

**Pull request merged**